flask/app/datasets.py

```diff
@@ -340,12 +340,30 @@ def delete_dataset(id: int):
     dataset = (
         models.Dataset.query.filter(models.Dataset.dataset_id == id)
         .options(joinedload(models.Dataset.analyses))
+        .options(joinedload(models.Dataset.groups))
         .first_or_404()
     )
+
+    # Remove permission groups with the corresponding dataset ID in groups_datasets table
+    dataset.groups = []
+
+    tissue_sample = (
+        models.TissueSample.query.filter(
+            models.TissueSample.tissue_sample_id == dataset.tissue_sample_id
+        )
+        .options(joinedload(models.TissueSample.datasets))
+        .first_or_404()
+    )
+
     if not dataset.analyses:
         try:
             db.session.delete(dataset)
             db.session.commit()
+
+            # Only delete tissue sample if it doesn't have any associated datasets left.
+            if len(tissue_sample.datasets) == 0:
```

Comment:
Does this field update without the prior commit? That would be preferable as this operation is achievable in one transaction.

Reply:
> I'm not quite sure what's going on with the reformatting in unrelated files.

I ran prettier --write on the react folder, which led to these changes. I probably should run it only on the files I change in the future!